src/ethereum-identity-provider.js

```diff
@@ -0,0 +1,55 @@
+'use strict'
+const IdentityProvider = require('./identity-provider')
+const { Wallet, utils } = require('ethers')
+const type = 'ethereum'
+
+class EthIdentityProvider extends IdentityProvider {
+  constructor(options) {
+    super()
+  }
+
+  // Returns the type of the identity provider
+  static get type () { return type }
+
+  // Returns the signer's id
+  async createId(options = {}) {
+    const wallet = options.wallet || await EthIdentityProvider.createWallet(options)
+    if (!wallet)
+      throw new Error(`wallet instance is required`)
+    this.wallet = wallet
+    return wallet.address
+  }
+
+  // Returns a signature of pubkeysignature
+  async signIdentity(pubKeySignature, options = {}) {
+    const wallet = options.wallet || this.wallet
+    if (!wallet)
+      throw new Error(`wallet is required`)
+
+    return await wallet.signMessage(pubKeySignature)
+  }
+
+  static async verifyIdentity (identity) {
+    // Verify that identity was signed by the id
+    const signerAddress = utils.verifyMessage(identity.publicKey + identity.signatures.id, identity.signatures.publicKey)
+    return (signerAddress === identity.id)
+  }
+
+  static async createWallet (options = {}) {
```

Comment:
It doesn't need to be but its useful for testing. Is there a reason not to?

Reply:
Reason not to keep it as public function is that a) it adds to the size of the API meaning there's another function that the users can call meaning more things that can break and b) because of a it means there's more for us to maintain/test and make sure to support in all possible ways. So if it's not _needed_ by the user, then default to private as that decreases the API surface which directly reduces complexity (to user) and support workload for maintainers :)